components/post_view/channel_intro_message/messages/default.tsx

```diff
@@ -43,35 +42,18 @@ const DefaultIntroMessage = ({
     teamIsGroupConstrained,
 }: Props) => {
     let teamInviteLink = null;
+    const channelIsArchived = channel.delete_at !== 0;
     const totalUsers = stats.total_users_count;
     const isPrivate = channel.type === Constants.PRIVATE_CHANNEL;
 
-    let setHeaderButton = null;
-    let boardCreateButton = null;
-    if (!isReadOnly) {
-        boardCreateButton = (
-            <BoardsButton
-                channel={channel}
-                boardComponent={boardComponent}
-            />
-        );
-        const children = (
-            <SetHeaderButton
-                channel={channel}
-            />
-        );
-        if (children) {
-            setHeaderButton = (
-                <ChannelPermissionGate
-                    teamId={channel.team_id}
-                    channelId={channel.id}
-                    permissions={[isPrivate ? Permissions.MANAGE_PRIVATE_CHANNEL_PROPERTIES : Permissions.MANAGE_PUBLIC_CHANNEL_PROPERTIES]}
-                >
-                    {children}
-                </ChannelPermissionGate>
-            );
-        }
-    }
+    const renderButtons = !isReadOnly && !channelIsArchived;
+    const boardCreateButton = renderButtons ? <BoardsButton boardComponent={boardComponent}/> : null;
+    const setHeaderButton = renderButtons ? (
+        <SetHeaderButton
+            channel={channel}
+            permissions={[isPrivate ? Permissions.MANAGE_PRIVATE_CHANNEL_PROPERTIES : Permissions.MANAGE_PUBLIC_CHANNEL_PROPERTIES]}
```

Comment:
to improve performance we should store this in a variable first, since adding in a literal will create a new instance of it every time this is run

Reply:
permissions was moved into the SetHeaderButton component